src/plugins/intel_cpu/src/transformation_pipeline.cpp

```diff
@@ -571,10 +570,34 @@ void Transformations::MainSnippets(void) {
     if (!isMHASupported) {
         snippetsManager.get_pass_config()->disable<ngraph::snippets::pass::TokenizeMHASnippets>();
     }
+
+    auto is_supported_matmul = [](const std::shared_ptr<const ov::op::v0::MatMul>& matmul) {
+        if (!matmul)
+            return false;
+        if (matmul->get_input_element_type(1) == ov::element::i8)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni);
+        if (matmul->get_input_element_type(0) == ov::element::bf16 &&
+            matmul->get_input_element_type(1) == ov::element::bf16)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16);
+        return true;
+    };
+
     if (snippetsMode != Config::SnippetsMode::IgnoreCallback) {
         snippetsManager.get_pass_config()->set_callback<ngraph::snippets::pass::TokenizeMHASnippets>(
-                [](const std::shared_ptr<const ov::Node>& n) -> bool {
-                    const auto pshape = n->get_output_partial_shape(0);
+                [this, is_supported_matmul](const std::shared_ptr<const ov::Node>& n) -> bool {
+                    if (this->enableLpt) {
+                        // Tranformation callback is called on MatMul1
+                        if (!is_supported_matmul(ov::as_type_ptr<const ov::op::v0::MatMul>(n)))
+                            return true;
+                        // Search for MatMul0
+                        auto parent = n->get_input_node_shared_ptr(0);
+                        while (!ov::is_type<ov::op::v0::MatMul>(parent)) {
+                            parent = parent->get_input_node_shared_ptr(0);
```

Comment:
It's really interesting why this works 🙃 It looks like I'm missing something, look:
1) We match on the first Matmul, it's supported so if on L591 passes
2) then we try to go upward until we meet another Matmul
3) But what happens is there is no other matmuls? Then we should visit a `Parameter` at last, which has no inputs, and catch smth like `out of range in get_argument(size_t index)`.
Could you please explain this point?))

Suggestion: It seems like a good idea to add a check on the inputs size in `while(...)`